### PR TITLE
Refactor concrete observable

### PR DESCRIPTION
**Pull request by M-Adoo**
1. Rename Observable to ObservableFromFn
2. Remove `Observable::new`, and add a same `new` function in `observable` mod to replace it

### CODE REVIEW

**Comment by M-Adoo** [comment]
A small  step for #62 , please code review @ishitatsuyuki and @helgoboss. And I invited you as a member to join this project, accept it to  improve it together If you guys think ok.

File: src/observable.rs [hunk 1]
```diff
@@ -30,24 +30,22 @@ macro impl_observable($t: ident) {
...
-    S: Observer<Item, Err>,
-  {
-    Self(subscribe, PhantomData)
-  }
+/// param `subscribe`: the function that is called when the Observable is
+/// initially subscribed to. This function is given a Subscriber, to which
+/// new values can be `next`ed, or an `error` method can be called to raise
+/// an error, or `complete` can be called to notify of a successful
+/// completion.
+pub fn new<F, Item, Err, S, U>(subscribe: F) -> ObservableFromFn<F, Item, Err>
```
Comment (ishitatsuyuki) [inline src/observable.rs]: I think `new` is probably too generic. Replace it with `from_fn`?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,17 @@
 ## [Unreleased](https://github.com/rxRust/rxRust/compare/v0.7.3...HEAD)
 
+### Features
+- **operator**: add `take_last` operator.
+
+### Refactor
+
+- **observable**: Observable split into many concrete type, not only use a Observable struct to wrap all, every observable creating function has a concrete type.
```
Comment (ishitatsuyuki) [inline CHANGELOG.md]: Concrete types means a type you can explicitly name, and using `impl Trait` will make a type unnameable. You might also want to update functions in `observable::trivial` to make them return their own structs, not ObserevableFromFn.

Comment (M-Adoo) [reply CHANGELOG.md]: I split this work into step: 
 - [x] rename Observable to ObservableFromFn #67
 - [ ] Observable creating functions return concrete class like ObservableRange .

This mr update the change log and  finish the first part. Just merge into  a refactor branch named `refactor-traits` not master.    I'll create a new mr to finish the second part.

File: src/observable.rs (hunk 1 above)
Comment (M-Adoo) [reply src/observable.rs]: `create`  better ?

Comment (ishitatsuyuki) [reply src/observable.rs]: Yeah I think we can go with `create`.

**Pull request merged**